build.ps1

```diff
@@ -35,76 +35,63 @@ try {
     }
 
     if ( -not (Test-Path .\OverlayPlugin.Updater\Resources\libcurl.dll)) {
-        echo "==> Building cURL..."
+        Write-Output "==> Building cURL..."
 
         mkdir .\OverlayPlugin.Updater\Resources
-        cd Thirdparty\curl\winbuild
+        Set-Location Thirdparty\curl\winbuild
 
-        echo "@call `"$VS_PATH\VC\Auxiliary\Build\vcvarsall.bat`" amd64"           | Out-File -Encoding ascii tmp_build.bat
-        echo "nmake /f Makefile.vc mode=dll VC=16 GEN_PDB=no DEBUG=no MACHINE=x64" | Out-File -Encoding ascii -Append tmp_build.bat
-        echo "@call `"$VS_PATH\VC\Auxiliary\Build\vcvarsall.bat`" x86"             | Out-File -Encoding ascii -Append tmp_build.bat
-        echo "nmake /f Makefile.vc mode=dll VC=16 GEN_PDB=no DEBUG=no MACHINE=x86" | Out-File -Encoding ascii -Append tmp_build.bat
+        Write-Output "@call `"$VS_PATH\VC\Auxiliary\Build\vcvarsall.bat`" amd64"           | Out-File -Encoding ascii tmp_build.bat
+        Write-Output "nmake /f Makefile.vc mode=dll VC=16 GEN_PDB=no DEBUG=no MACHINE=x64" | Out-File -Encoding ascii -Append tmp_build.bat
+        Write-Output "@call `"$VS_PATH\VC\Auxiliary\Build\vcvarsall.bat`" x86"             | Out-File -Encoding ascii -Append tmp_build.bat
+        Write-Output "nmake /f Makefile.vc mode=dll VC=16 GEN_PDB=no DEBUG=no MACHINE=x86" | Out-File -Encoding ascii -Append tmp_build.bat
 
         cmd "/c" "tmp_build.bat"
-        sleep 3
-        del tmp_build.bat
+        Start-Sleep 3
+        Remove-Item tmp_build.bat
 
-        cd ..\builds
-        copy .\libcurl-vc16-x64-release-dll-ipv6-sspi-winssl\bin\libcurl.dll ..\..\..\OverlayPlugin.Updater\Resources\libcurl-x64.dll
-        copy .\libcurl-vc16-x86-release-dll-ipv6-sspi-winssl\bin\libcurl.dll ..\..\..\OverlayPlugin.Updater\Resources\libcurl.dll
+        Set-Location ..\builds
+        Copy-Item .\libcurl-vc16-x64-release-dll-ipv6-sspi-winssl\bin\libcurl.dll ..\..\..\OverlayPlugin.Updater\Resources\libcurl-x64.dll
+        Copy-Item .\libcurl-vc16-x86-release-dll-ipv6-sspi-winssl\bin\libcurl.dll ..\..\..\OverlayPlugin.Updater\Resources\libcurl.dll
 
-        cd ..\..\..
+        Set-Location ..\..\..
     }
 
     if ($ci) {
-        echo "==> Continuous integration flag set. Building Debug..."
-        msbuild -p:Configuration=Debug -p:Platform=x64 "OverlayPlugin.sln" -t:Restore
-        msbuild -p:Configuration=Debug -p:Platform=x64 "OverlayPlugin.sln"    
+        Write-Output "==> Continuous integration flag set. Building Debug..."
+        dotnet publish -c debug  
     }
 
-    echo "==> Building..."
+    Write-Output "==> Building..."
 
-    msbuild -p:Configuration=Release -p:Platform=x64 "OverlayPlugin.sln" -t:Restore
-    msbuild -p:Configuration=Release -p:Platform=x64 "OverlayPlugin.sln"
+    dotnet publish -c release
```

Comment:
I'm not sure on the exact semantics of `dotnet publish` off the top of my head, so I just want to double check. The lines it's replacing are doing a nuget package fetch/restore (via `-t:Restore`), then a build. Are both of those accomplished with the single command?

Reply:
Yeah they are unless you add `--no-restore` to disable implicit restore